File: solutions/csharp/word-count/1/WordCount.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

public static class WordCount
{
    public static IDictionary<string, int> CountWords(string phrase)
    {
        phrase = phrase.ToLower();
        phrase = Regex.Replace(phrase,"[^a-z0-9 ']+", " ");
        phrase = phrase.Trim();
        phrase =phrase.Replace("  "," ");
        phrase =phrase.Replace("  "," ");

        string[] words = phrase.Split(new char[0]);
        IDictionary<string, int> wc = new Dictionary<string, int>();
        char[] apos = {'\'','"'};
        foreach(string word in words) {
           string tword = word.Trim(apos);
           if(wc.ContainsKey(tword)) wc[tword]++;
           else wc.Add(tword.ToLower(),1);
        }
        return wc;
        //throw new NotImplementedException("You need to implement this function.");
    }
}

File: solutions/python/word-count/1/word_count.py
import re
def word_count(phrase):
    phrase = phrase.lower()
    words = re.findall(r"[a-z0-9']+",phrase)
    print (words)
    words2 = []
    for word in words:
        if word[0] in "'"+'"' and word[-1] in "'"+'"':
            words2.append(word[1:-1])
        else:
            words2.append(word)   
    wc = {word: words2.count(word) for word in words2}
    return wc
    

print(word_count('hey,my_spacebar_is_broken.'))